packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -332,7 +332,7 @@ class ScaffoldState extends State<Scaffold> {
         bottomSheet,
         completer,
         () => entry.remove(),
-        setState
+        (VoidCallback fn) { bottomSheetKey.currentState?.setState(fn); }
```

Comment:
I've filed a related issue: Rebuilding the Scaffold doesn't rebuild its BottomSheet
https://github.com/flutter/flutter/issues/2115